### PR TITLE
Extended calculated statistics

### PR DESCRIPTION
**Pull request by janniclas**
This PR adds statistics for the following instruction counts:

-  size_t Branches = 0;
- size_t GetElementPtrs = 0;
- size_t Terminators = 0;
- size_t PhiNodes = 0;

Further some dead code was removed.

### CODE REVIEW

File: include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h [hunk 1]
```diff
@@ -45,6 +45,10 @@ class GeneralStatistics {
   size_t LoadInstructions = 0;
   size_t MemIntrinsics = 0;
   size_t GlobalPointers = 0;
+  size_t Branches = 0;
+  size_t GetElementPtrs = 0;
+  size_t Terminators = 0;
```
Comment (fabianbs96) [inline include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h:50]: It seems to be redundant to count terminators as there is always exactly one terminator per basic-block

**Review by fabianbs96 — COMMENTED** [review]
Looks already quite good; 
maybe you want to print the extended statistics from phasar-llvm -S as well

File: include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h (hunk 1 above)
Comment (janniclas) [reply include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h:50]: good point, didn't think about that.

**Review by janniclas — COMMENTED** [review]
removed separate counter for terminators

**Comment by janniclas** [comment]
@fabianbs96 I included the changes we discussed in regards of exposing the complete GeneralStatistics from the IRDB class, as well as overloading the << operator in the statistics class to ease printing and cleanup the code in phasar-llvm.cpp

**Pull request merged**